pkg/util/exclude_kubernetes.go

```diff
@@ -173,3 +214,62 @@ func UnExcludeKubernetesResource(fs afero.Afero, basePath string, unExcludedReso
 
 	return fmt.Errorf("unable to find resource %s in %s or its bases", unExcludedResource, basePath)
 }
+
+func getKustomization(fs afero.Afero, basePath string) (*types.Kustomization, error) {
+	exists, err := fs.Exists(filepath.Join(basePath, "kustomization.yaml"))
+	if err != nil {
+		return nil, errors.Wrapf(err, "check kustomization yaml in %s", basePath)
+	}
+	if !exists {
+		return nil, fmt.Errorf("kustomization in %s does not exist", basePath)
+	}
+
+	kustomizeYaml, err := fs.ReadFile(filepath.Join(basePath, "kustomization.yaml"))
+	if err != nil {
+		return nil, errors.Wrapf(err, "read kustomization yaml in %s", basePath)
+	}
+
+	kustomization := types.Kustomization{}
+	err = yaml.Unmarshal(kustomizeYaml, &kustomization)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshal kustomization yaml from %s", basePath)
+	}
+	return &kustomization, nil
+}
+
+func writeKustomization(fs afero.Afero, basePath string, kustomization *types.Kustomization) error {
+	kustomizeYaml, err := yaml.Marshal(kustomization)
+	if err != nil {
+		return errors.Wrapf(err, "marshal kustomization yaml from %s", basePath)
+	}
+
+	err = fs.WriteFile(filepath.Join(basePath, "kustomization.yaml"), kustomizeYaml, 0644)
+	if err != nil {
+		return errors.Wrapf(err, "write kustomization yaml to %s", basePath)
+	}
+	return nil
+}
+
+// checks if the merge patch file at a given path matches the provided resource
+func mergePatchMatches(fs afero.Afero, basePath string, mergePatch string, excludedResource resid.ResId) (bool, error) {
+	// read contents of resource and convert it to ResID form
+	patchBytes, err := fs.ReadFile(filepath.Join(basePath, string(mergePatch)))
+	if err != nil {
+		return false, errors.Wrapf(err, "read %s in %s to exclude patches for %s", string(mergePatch), basePath, excludedResource.String())
```

Comment:
doesnt `%s` essentially call `string()` and `.String()`?
https://tour.golang.org/methods/17

Reply:
I prefer to be explicit
And not to rely on type discovery...